steps/measure.py

```diff
@@ -91,3 +93,29 @@ def evaluate_ansatz_based_cost_function(
     value_estimate = cost_function(ansatz_parameters)
 
     save_value_estimate(value_estimate, "value_estimate.json")
+
+def hamiltonian_analysis(
+    qubit_operator: str,
+    decomposition_method: str = "greedy",
+    expectation_values: str = "None",
+):
+    operator = load_qubit_operator(qubit_operator)
+    if decomposition_method != "greedy-sorted" and decomposition_method != "greedy":
+        raise Exception(f'Decomposition method {decomposition_method} is not supported')
```

Comment:
As previously, we should probably raise `ValueError` here.

Reply:
Fair point, I replaced `Exception` with `ValueError` in both cases